client/my-sites/email/email-providers-stacked-comparison/email-provider-stacked-card/email-provider-generic-form.tsx

```diff
@@ -0,0 +1,868 @@
+								} }
+								onBlur={ () => {
+									setLastNameFieldTouched( wasValidated );
+								} }
+								onKeyUp={ onReturnKeyPress }
+							/>
+						</LabelWrapper>
+					) }
+
+					{ hasLastNameError && <FormInputValidation text={ lastNameError } isError /> }
+				</div>
+			</FormFieldset>
+
+			{ ! hiddenFields.some(
+				( field ) => field === GENERIC_EMAIL_FORM_ALTERNATIVE_EMAIL_FIELD
+			) && (
+				<FormFieldset className="email-provider-generic-form__form-fieldset">
+					<div className="email-provider-generic-form__new-user-name-container">
+						<LabelWrapper
+							label={ translate( 'Password reset email address', {
+								comment: 'This is the email address we will send password reset emails to',
+							} ) }
+						>
+							<FormTextInput
+								value={ alternativeEmail }
+								isError={ hasAlternativeEmailError }
+								onChange={ ( event: ChangeEvent< HTMLInputElement > ) => {
+									onUserValueChange(
+										GENERIC_EMAIL_FORM_ALTERNATIVE_EMAIL_FIELD,
+										event.target.value
+									);
+								} }
+								onBlur={ () => {
+									setAlternativeEmailFieldTouched( wasValidated );
+								} }
+								onKeyUp={ onReturnKeyPress }
+							/>
+						</LabelWrapper>
+						{ hasAlternativeEmailError && (
+							<FormInputValidation text={ alternativeEmailError } isError />
+						) }
+					</div>
+				</FormFieldset>
+			) }
+
+			<FormFieldset className="email-provider-generic-form__form-fieldset">
+				<div className="email-provider-generic-form__new-user-email-container">
+					<div className="email-provider-generic-form__new-user-email">
+						{ domains.length > 1 ? renderMultiDomain() : renderSingleDomain() }
+					</div>
+
+					{ hasMailBoxError && <FormInputValidation text={ mailBoxError } isError /> }
+				</div>
+
+				<div className="email-provider-generic-form__new-user-password-container">
+					<LabelWrapper label={ translate( 'Password' ) }>
+						<FormPasswordInput
+							autoCapitalize="off"
+							autoCorrect="off"
+							value={ password }
+							maxLength={ 100 }
+							isError={ hasPasswordError }
+							onChange={ ( event: ChangeEvent< HTMLInputElement > ) => {
+								onUserValueChange( 'password', event.target.value );
+							} }
+							onBlur={ () => {
+								setPasswordFieldTouched( wasValidated );
+							} }
+							onKeyUp={ onReturnKeyPress }
+						/>
+					</LabelWrapper>
+
+					{ hasPasswordError && <FormInputValidation text={ passwordError } isError /> }
+				</div>
+				{ showTrashButton && (
+					<Button
+						className="email-provider-generic-form__new-user-remove-user-button"
+						onClick={ onUserRemove }
+					>
+						<Gridicon icon="trash" />
+						<span>{ translate( 'Remove this mailbox' ) }</span>
+					</Button>
+				) }
+			</FormFieldset>
+		</div>
+	);
+};
+
+interface EmailProviderGenericFormProps {
+	autoFocus?: boolean;
+	children?: ReactNode;
+	domains?: any;
```

Comment:
`domains` should be typed as an array of `SiteDomain` or `ResponseDomain` (with the latter being more complete at the moment).